### PR TITLE
Show affected objects in the object detail for failed dependencies

### LINKED ISSUE #1064: Show a list of affected children in the host/service detail if it has a problem and is part of a dependency
As a complement to #1050, we want to list the most severe children that are affected by a parent in its detail area.

**It should look like this:**

## Notes

* The list may contain objects of type *host*, *service* and *redundancy group*
* It is shown if any of the dependencies, the object is the parent of, have failed
* Only direct children and only those whose dependency has failed are listed
  * Results are ordered by severity, if applicable
  * Restrictions apply as usual
* Each item links to its detail area
  * Redundancy groups have none yet (See #1056)
* Results are limited to 5 in total
  * A link to all results if there are more is shown (See #1063)
* The list is entirely replaceable by hooks of type `ObjectDetailExtensionHook`
  * It is located in the `RELATED` section and uses the position 510

### PR DESCRIPTION
**Pull request by raviks789**
resolves #1064

### CODE REVIEW

**Review by sukhwinder33445 — CHANGES_REQUESTED** [review]
The list is not limited to 5 items, as required.

File: library/Icingadb/Widget/Detail/ObjectDetail.php [hunk 1]
```diff
@@ -37,6 +38,7 @@
 use Icinga\Module\Icingadb\Util\PluginOutput;
 use Icinga\Module\Icingadb\Widget\ItemList\DowntimeList;
 use Icinga\Module\Icingadb\Widget\ShowMore;
+use ipl\Sql\Select;
```
Comment (sukhwinder33445) [inline library/Icingadb/Widget/Detail/ObjectDetail.php:41]: This is unused.

File: library/Icingadb/Widget/Detail/ObjectDetail.php [hunk 2]
```diff
@@ -648,4 +650,47 @@ protected function createRootProblems(): ?array
             new DependencyNodeList($rootProblems)
         ];
     }
+
+    protected function createAffectedObjects()
```
Comment (sukhwinder33445) [inline library/Icingadb/Widget/Detail/ObjectDetail.php:654]: Please add the return type and PHPDoc.

File: library/Icingadb/Widget/Detail/ObjectDetail.php [hunk 3]
```diff
@@ -656,4 +657,56 @@ protected function createRootProblems(): ?array
...
+                'service.host.state'
+            ])
+            ->setResultSetClass(VolatileStateResults::class)
+            ->limit(5)
+            ->orderBy([
+                'host.state.severity',
+                'host.state.last_state_change',
+                'service.state.severity',
+                'service.state.last_state_change',
+                'redundancy_group.state.last_state_change'
```
Comment (nilmerg) [inline library/Icingadb/Widget/Detail/ObjectDetail.php:694]: Please combine this as well with the group's state.

File: library/Icingadb/Widget/Detail/ObjectDetail.php [hunk 4]
```diff
@@ -656,4 +657,56 @@ protected function createRootProblems(): ?array
...
+     *
+     * @return ?BaseHtmlElement[]
+     */
+    protected function createAffectedObjects(): ?array
+    {
+        if (! $this->object->state->affects_children) {
+            return null;
+        }
+
+        $affectedObjects = DependencyNode::on($this->getDb())
```
Comment (nilmerg) [inline library/Icingadb/Widget/Detail/ObjectDetail.php:672]: This lists also objects which are not affected. The issue notes:

> Only direct children **and only those whose dependency has failed** are listed

**Review by nilmerg — CHANGES_REQUESTED** [review]

**Review by nilmerg — APPROVED** [review]
LGTM!

**Pull request merged**